### ISSUE TITLE
Need to be able to specify checks required for merge.

### ISSUE THREAD

**Issue by chriseyre2000** [dependabot/feedback#559]
It would be useful if we could specify the minimum required checks before a merge happens.

We use Jenkins for unit test and our platform team have been experimenting with snyk.
The problem is if snyk responds first then a build can be merged that has failing tests.

We either need to be able to do this in the UI or in the config file.
All I really need to to be able to see that the unit tests have passed.
Snyk should also be able to fail the build.

**Comment by greysteil** [comment]
Hmmm, is it possible to do this on GitHub already? I've seen PRs with statuses that say "required" on them. Perhaps Dependabot could piggy back off of that UI...

**Comment by chriseyre2000** [comment]
Jenkins reports success in a different way. What I would like at a minimum is to have snyk is not enough to merge on it's own. (ie. Snyk can fail a build but it is not enough to pass on it's own).

**Comment by chriseyre2000** [comment]
We now know more about this.

There is a race condition between Jenkins reporting that testing has started and Snyk running checks. If Snyk wins it will merge the change even if the tests fail.

**Comment by chriseyre2000** [comment]
I heard that dependabot keeps a history of known checks and will wait for them.
It would be useful for us to be able to specify these checks ourselves.

**Comment by greysteil** [comment]
Sadly we don't do that anymore - it was a nightmare to keep up-to-date, and caused a lot of "why didn't this merge?" inquiries. We switched to listening for  check runs which broadly fixed things.

So in this case it sounds like the trouble is that the Snyk check can finish so fast that the Jenkins one hasn't even told GitHub that it's processing? I think just adding a delay on the Dependabot side might be the way to fix that, assuming Jenkins responds (to say it's enqueued / processing) in a relatively timely way (like, within a minute).

**Comment by chriseyre2000** [comment]
That might help, but it would be better if we had an (optional) list of checks that (if present) we expect to run.

**Comment by greysteil** [comment]
Yeah, trouble is that sounds like it should definitely be a GitHub feature, rather than something that lives in Dependabot (especially since we're working to integrate Dependabot into GitHub), and figuring out how a UI for that would work is a bunch harder...

**Comment by greysteil** [comment]
OK, I've implemented a minimum wait before Dependabot merges a PR. It's set to 60 seconds, which hopefully should be long enough for your check runs to report that they've started.

**Comment by chriseyre2000** [comment]
Thanks

**Comment by greysteil** [comment]
No worries - thanks for the feedback! It's how we get better :-)

**Comment by eps1lon** [comment]
We have a couple of checks marked as not required in our repository. These tasks are usually informative only or are pretty flaky (e.g. netlify status checks). Since we still explicitly tell dependabot to "squash and merge" it would be sufficient of the bot ignores pending status checks that arent' required.

Regarding how this would be implemented: 
The GitHub API has a endpoint for branch protection rules (which is how required status checks are implemented). You could query those and check if status in the branch protection rules currently passes.
